artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -742,6 +756,25 @@ public void addConsumer(ConsumerInfo info) throws Exception {
       }
    }
 
+   public void setConnectionEntry(ConnectionEntry connectionEntry) {
+      this.connectionEntry = connectionEntry;
+   }
+
+   public void setUpTtl(final long inactivityDuration, final long inactivityDurationInitialDelay, final boolean useKeepAlive) {
+      this.useKeepAlive = useKeepAlive;
+      this.maxInactivityDuration = inactivityDuration;
+
+      protocolManager.getScheduledPool().schedule(new Runnable() {
```

Comment:
@gaohoward Really? I thought they were different like AMQP.

Why do you need the scheduled executor then? just to set the variable?


      protocolManager.getScheduledPool().schedule(new Runnable() {
         @Override
         public void run() {
            if (inactivityDuration >= 0) {
               connectionEntry.ttl = inactivityDuration;
            }
         }
      }, inactivityDurationInitialDelay, TimeUnit.MILLISECONDS);